src/test/java/com/powsybl/openloadflow/network/VoltageControlNetworkFactory.java

```diff
@@ -1070,4 +1070,159 @@ public static Network createWithShuntAndGeneratorVoltageControl() {
+                .setBus1("b4")
+                .setBus2("b5")
+                .setOpen(true)
+                .add();
+
+        b1.getVoltageLevel()
+                .newGenerator()
+                .setId("g1")
+                .setBus("b1")
+                .setConnectableBus("b1")
+                .setEnergySource(EnergySource.THERMAL)
+                .setMinP(0)
+                .setMaxP(200)
+                .setTargetP(100)
+                .setTargetV(400)
+                .setRegulatingTerminal(l5.getTerminal())
+                .setVoltageRegulatorOn(true)
+                .add();
+        b2.getVoltageLevel().newShuntCompensator()
+                .setId("SHUNT2")
+                .setBus("b2")
+                .setConnectableBus("b2")
+                .setSectionCount(0)
+                .setVoltageRegulatorOn(false)
+                .setRegulatingTerminal(l4.getTerminal())
+                .setTargetV(380)
+                .setTargetDeadband(5.0)
+                .newLinearModel()
+                .setMaximumSectionCount(50)
+                .setBPerSection(-1E-2)
+                .setGPerSection(0.0)
+                .add()
+                .add();
+        b3.getVoltageLevel().newShuntCompensator()
+                .setId("SHUNT3")
+                .setBus("b3")
+                .setConnectableBus("b3")
+                .setSectionCount(0)
+                .setVoltageRegulatorOn(true)
+                .setRegulatingTerminal(l4.getTerminal())
+                .setTargetV(380)
+                .setTargetDeadband(5.0)
+                .newLinearModel()
+                .setMaximumSectionCount(50)
+                .setBPerSection(-1E-2)
+                .setGPerSection(0.0)
+                .add()
+                .add();
+        s.newTwoWindingsTransformer()
+                .setId("tr1")
+                .setBus1(b1.getId())
+                .setConnectableBus1(b1.getId())
+                .setBus2(b5.getId())
+                .setConnectableBus2(b5.getId())
+                .setRatedU1(20.5)
+                .setRatedU2(399)
+                .setR(1)
+                .setX(30)
+                .add();
+        s.newTwoWindingsTransformer()
+                .setId("tr2")
+                .setBus1(b2.getId())
+                .setConnectableBus1(b2.getId())
+                .setBus2(b4.getId())
+                .setConnectableBus2(b4.getId())
+                .setRatedU1(20.5)
+                .setRatedU2(399)
+                .setR(1)
+                .setX(30)
+                .add();
+        s.newTwoWindingsTransformer()
+                .setId("tr3")
+                .setBus1(b3.getId())
+                .setConnectableBus1(b3.getId())
+                .setBus2(b4.getId())
+                .setConnectableBus2(b4.getId())
+                .setRatedU1(20.5)
+                .setRatedU2(399)
+                .setR(1)
+                .setX(30)
+                .add();
+        createLine(network, b1, b2, "l12", 0.00);
```

Comment:
This line is not on the network diagram. Is it on purpose?

Reply:
I am going to fix the scheme but I need this line to have one synchronous component (and not two).